tripy/tripy/frontend/trace/ops/slice.py

```diff
@@ -271,19 +272,26 @@ def slice_helper(tensor, *slice_params: TensorLike):
     frame_index = -1
     assert slice_params
 
+    # Internal WAR: the constraints decorator is applied before the function registry decorator, so in the constraints tests,
+    # we will not find the Tensor.__getitem__ decorator. We can use a fallback in that case.
```

Comment:
This was a bit of a wacky interaction with #333. FWIW, the previous approach of "find the call to `__getitem__` and skip one more stack entry" would avoid the need for a WAR here. The fundamental issue is the order in which decorators are applied.